File: music-cloud-web/src/main/web/src/app/dto/dialog.ts
import { LocalDate } from './local-date';
import { User } from './user';
import { Message } from './message';

export class Dialog {
    private _id: number;
    private _name: string;
    private _time: LocalDate;
    private _users: User[] = [];
    private _messages: Message[] = [];

    public static createEmptyDialog() {
        return new Dialog({
            id: -1,
            name: '',
            time: {},
            users: [],
            messages: []
        });
    }


    constructor(data: any) {
        this._id = data.id;
        this._name = data.name;
        this._time = new LocalDate(data.time);
        data.users.forEach(user => {
            this._users.push(new User(user));
        });
        data.messages.forEach(message => {
            this._messages.push(new Message(message));
        });
    }

    get id(): number {
        return this._id;
    }

    set id(value: number) {
        this._id = value;
    }

    get name(): string {
        return this._name;
    }

    set name(value: string) {
        this._name = value;
    }

    get time(): LocalDate {
        return this._time;
    }

    set time(value: LocalDate) {
        this._time = value;
    }

    get users(): User[] {
        return this._users;
    }

    set users(value: User[]) {
        this._users = value;
    }

    get messages(): Message[] {
        return this._messages;
    }

    set messages(value: Message[]) {
        this._messages = value;
    }
}

File: music-cloud-server/src/main/java/com/kulygin/musiccloud/domain/Dialog.java
package com.kulygin.musiccloud.domain;

import lombok.*;

import javax.persistence.*;
import java.time.LocalDateTime;
import java.util.Set;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
@Entity
@Table(schema = "public")
@EqualsAndHashCode(of = { "id"})
@ToString(exclude = {"users", "messages"})
public class Dialog {
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Long id;
    private String name;
    private LocalDateTime time;

    @ManyToMany(fetch = FetchType.LAZY)
    @JoinTable(name = "user_has_dialog",
            joinColumns = {@JoinColumn(name = "dialog_id")},
            inverseJoinColumns = {@JoinColumn(name = "user_id")})
    private Set<User> users;

    @ManyToMany(fetch = FetchType.LAZY)
    @JoinTable(name = "message_has_dialog",
            joinColumns = {@JoinColumn(name = "dialog_id")},
            inverseJoinColumns = {@JoinColumn(name = "message_id")})
    private Set<Message> messages;
}
